flytekit/core/inference.py

```diff
@@ -0,0 +1,196 @@
+"""
+=========
+Inference
+=========
+
+.. currentmodule:: flytekit.core.inference
```

Comment:
Having a `flytekit.core.inference` feels very ML/AI leaning, should this be a plugin?

Reply:
hmm... what about `serve`? i don't think this should be a plugin as it's just a pod template wrapper.

Reply:
It's a pod template wrapper that uses the huggingface API to download a model, which is very AI focused. Although I think the feature is interesting, I do not know if it fits flytekit core.

@eapolinario What do you think?

Reply:
> It's a pod template wrapper that uses the huggingface API to download a model, which is very AI focused. Although I think the feature is interesting, I do not know if it fits flytekit core.
> 
> @eapolinario What do you think?

Agreed. What was the reasoning to put this in core, @samhita-alla ? 



> what about `serve`? i don't think this should be a plugin as it's just a pod template wrapper.

Even if it's just a pod template wrapper, it's a *very* useful wrapper.

What was the reasoning behind putting this in core, @samhita-alla ?

Reply:
@eapolinario given that this implementation requires no additional libraries and functions as a wrapper rather than a plugin, wouldn't it be more appropriate to include it in flytekit core? also, `flytekit.core` was Ketan's suggestion.

Reply:
I disagree with adding this code to Flytekit core simply because it doesn't introduce extra dependencies. We already have an extension mechanism through Flytekit plugins. As @thomasjpfan pointed out, flytekit doesn't include ML-specific code yet and bringing NIM to core creates a precedent and puts more burden on the maintainers, for example, what do we do when we receive external contributions?

So maybe, let's take a step back and list what are the drawbacks of creating a Nim plugin instead?

@kumare3, can you chime in here?

Reply:
i'll move nim to flytekit plugins